### PR TITLE
swapd: from_height, scan wallet for some blocks in the past

### PR DESCRIPTION
**Pull request by zkao**

### CODE REVIEW

File: src/swapd/runtime.rs [hunk 1]
```diff
@@ -795,7 +795,7 @@ impl SyncerState {
     }
     fn watch_addr_btc(&mut self, script_pubkey: Script, tx_label: TxLabel) -> Task {
         let id = self.tasks.new_taskid();
-        let from_height = self.height(Coin::Bitcoin);
+        let from_height = self.height(Coin::Bitcoin) - 5;
```
Comment (TheCharlatan) [inline src/swapd/runtime.rs:798]: This can underflow, maybe do `cmp::min(self.height(Coin::Bitcoin) - 5, self.height(Coin::Bitcoin))`

Comment (zkao) [reply src/swapd/runtime.rs:798]: true

Comment (zkao) [reply src/swapd/runtime.rs]: c95e90e

File: src/swapd/runtime.rs [hunk 2]
```diff
@@ -819,6 +819,12 @@ impl SyncerState {
     fn is_watched_addr(&self, tx_label: &TxLabel) -> bool {
         self.tasks.watched_addrs.values().any(|tx| tx == tx_label)
     }
+    fn from_height(&self, coin: Coin, delta: u64) -> u64 {
+        let height = self.height(coin);
+        let delta = if height > delta { delta } else { height };
+        height - delta
```
Comment (Lederstrumpf) [inline src/swapd/runtime.rs:825]: if height <= delta, then from_height = 0

**Review by Lederstrumpf — APPROVED** [review]

**Pull request merged**